index.html

```diff
@@ -86,34 +86,124 @@
+                    } else {
+                        read_storage = true;
                         if (editor_exists) {
-                            editor.setValue(project_src, -1);
+                            loadCode("# Couldn't load gist  (http != 200).\n# Please try again later.\n\n")
                         }
+                        return null;
                     }
-                    else {
-                        read_storage = true;
+
+                }).then(t => {
+                    if (t) {
+
+                        console.log("Gist content fetched, setting code...");
+
+                        // Add a small prefix to inform the user whether the gist was loaded from the server or from local storage
+                        editor_text = `# Content loaded from Github gist on ${new Date().toLocaleString()}\n# https://gist.github.com/${gist_slug.replace("-", "/")}\n\n${t}`;
+
+                        console.log(editor_text);
+
                         if (editor_exists) {
-                            loadCode()
+                            editor.setValue(editor_text, -1);
+                            saveCode();
+                        } else {
+
+                            console.log("The editor hasn't loaded yet. Retrying every second.");
+
+                            window.loadInGistInterval = setInterval(() => {
+                                if (editor_exists) {
+                                    console.log("The editor loaded, setting code...");
+                                    editor.setValue(editor_text, -1);
+                                    saveCode();
+                                    clearInterval(window.loadInGistInterval);
+                                } else {
+                                    console.log("The editor hasn't loaded yet.");
+                                }
+                            }, 100);
+
                         }
                     }
-                };
-                client.send();
+                }).catch(e => {
+
+                    read_storage = true;
+                    if (editor_exists) {
+                        loadCode("# Couldn't load gist (fetch error).\n# Please try again later.\n\n")
+                    }
+
+                });
+
+            } else {
+                read_storage = true;
+                if (editor_exists) {
+                    loadCode("# Couldn't load gist (invalid gist slug).\n# Please try again later.\n\n")
+                }
             }
-            var url = window.location.href;
-            var newUrl = url.substring(url.lastIndexOf('/') + 1).split("?")[0];
-            window.history.pushState("object or string", "Title", "" + newUrl);
+
         }
+
+        // var url = window.location.href;
+        // var newUrl = url.substring(url.lastIndexOf('/') + 1).split("?")[0];
+        // window.history.pushState("object or string", "Title", "" + newUrl);
+
+        function loadGistFromUrl() {
+

```

Comment:
nit: the new lines before and after the function body are a little excessive

Reply:
Yep, removed those